packages/client/src/StreamVideoClient.ts

```diff
@@ -120,13 +122,51 @@ export class StreamVideoClient {
       const user = apiKeyOrArgs.user;
       const token = apiKeyOrArgs.token || apiKeyOrArgs.tokenProvider;
       if (user) {
+        let id = user.id;
+        if (user.type === 'anonymous') {
+          id = '!anon';
+        }
+        if (id && StreamVideoClient._instanceMap.has(id)) {
+          this.logger(
+            'warn',
+            'A StreamVideoClient already exists for the given user ID; Prefer using getOrCreateInstance method',
+          );
+        }
         this.connectUser(user, token).catch((err) => {
           this.logger('error', 'Failed to connect', err);
         });
       }
     }
   }
 
+  public static getOrCreateInstance(args: {
+    apiKey: string;
+    user: User;
+    tokenProvider: TokenProvider;
```

Comment:
I made user and tokenProvider to be required here.. that makes it simpler


client has to be tied to user-id so User has to be required

alternative to not making tokenProvider required is strange to me as User always ties with token when we connect the user